src/main/java/server/Request.java

```diff
@@ -0,0 +1,17 @@
+package server;
+
+import java.util.HashMap;
+
+public class Request {
+    private HashMap parsedRequest;
```

Comment:
This would fall under the dependency inversion principle, but you should be depending on the abstract interfaces as much as possible, unless there is a specific reason this needs to be a `HashMap`. 
This is also where the additional type information will come in handy. As an example, you can specify that a `Request` can be constructed with any type of `Map` as long as the keys are `String` and the values are `Object`.

Reply:
So use the highest level of abstraction that I can get away with in my declarations. That makes sense, thanks!